MidasCivil_Adapter/Convert/ToMidasCivil/Elements/FromFEMesh.cs

```diff
@@ -48,31 +54,44 @@ public static string FromFEMesh(this FEMesh feMesh)
                 }
             }
-            if (feMesh.Nodes.Count > 4)
-            {
-                BH.Engine.Reflection.Compute.RecordError("Cannot push mesh with more than 4 nodes");
-            }
 
-            if (feMesh.Nodes.Count == 4)
+            foreach (FEMeshFace meshFace in feMesh.Faces)
             {
-                midasElement = (feMesh.AdapterId<string>(typeof(MidasCivilId)) + ",PLATE," +
-                materialId + "," +
-                sectionPropertyId + "," +
-              feMesh.Nodes[nodeIndices[0]].AdapterId<string>(typeof(MidasCivilId)) + "," +
-              feMesh.Nodes[nodeIndices[1]].AdapterId<string>(typeof(MidasCivilId)) + "," +
-              feMesh.Nodes[nodeIndices[2]].AdapterId<string>(typeof(MidasCivilId)) + "," +
-              feMesh.Nodes[nodeIndices[3]].AdapterId<string>(typeof(MidasCivilId)) + ",1,0");
-            }
-            else
-            {
-                midasElement = (feMesh.AdapterId<string>(typeof(MidasCivilId)) + ",PLATE," +
-                materialId + "," +
-                sectionPropertyId + "," +
-             feMesh.Nodes[nodeIndices[0]].AdapterId<string>(typeof(MidasCivilId)) + "," +
-             feMesh.Nodes[nodeIndices[1]].AdapterId<string>(typeof(MidasCivilId)) + "," +
-             feMesh.Nodes[nodeIndices[2]].AdapterId<string>(typeof(MidasCivilId)) + ",0,1,0");
+                int node0 = meshFace.NodeListIndices[0];
+                int node1 = meshFace.NodeListIndices[1];
+                int node2 = meshFace.NodeListIndices[2];
+
+                if (meshFace.NodeListIndices.Count < 3)
+                {
+                    BH.Engine.Reflection.Compute.RecordError("Cannot push mesh with less than 3 nodes");
+                }
+
+                if (meshFace.NodeListIndices.Count == 4)
+                {
+                    int node3 = meshFace.NodeListIndices[3];
+                    midasElement = (index + ",PLATE," +
+                    materialId + "," +
+                    sectionPropertyId + "," +
+                    feMesh.Nodes[node0].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                    feMesh.Nodes[node1].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                    feMesh.Nodes[node2].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                    feMesh.Nodes[node3].AdapterId<string>(typeof(MidasCivilId)) + ",1,0");
+                }
+                else if (meshFace.NodeListIndices.Count == 3)
+                {
+                    midasElement = (index + ",PLATE," +
+                    materialId + "," +
+                    sectionPropertyId + "," +
+                    feMesh.Nodes[node0].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                    feMesh.Nodes[node1].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                    feMesh.Nodes[node2].AdapterId<string>(typeof(MidasCivilId)) + ",0,1,0");
+                }
+
```

Comment:
```suggestion
                meshFace.SetAdapterId(typeof(MidasCivilId), index);
```

Can we add this in to make sure that the `AdapterId` is correct on the BHoM object.